gqlalchemy/memgraph.py

```diff
@@ -543,3 +556,146 @@ def create_relationship(self, relationship: Relationship) -> Optional[Relationsh
+        """Parse signature and make a dictionary for every argument and return.
+
+        For instance, if a query module signature is:
+        dummy_module.dummy(lst :: LIST OF STRING, num = 3 :: NUMBER) :: (ret :: STRING)
+        the method should return a list of arguments:
+        [{"name": "lst", "type": "LIST OF STRING"}, {"name": "num", "type": "NUMBER", "default": 3}]
+        and a list of returns:
+        [{"name": "ret", "type": "STRING"}]
+
+        Dictionary consists of fields: "name" - argument name, "type" - data
+        type of argument and "default" where default argument value is given
+
+        Args:
+            signature: module signature as returned by Cypher CALL operation
+        """
+        end_arguments_parantheses = signature.index(")")
+        arguments_field = signature[signature.index("(") + 1 : end_arguments_parantheses]
+        returns_field = signature[
+            signature.index("(", end_arguments_parantheses) + 1 : signature.index(")", end_arguments_parantheses + 1)
+        ]
+
+        arguments = self._parse_field(arguments_field.strip())
+        returns = self._parse_field(returns_field.strip())
+
+        return arguments, returns
+
+    def _parse_field(self, vars_field: str) -> List[Dict]:
+        """Parse a field of arguments or returns from module signature
+
+        Args:
+            vars_field: signature field inside parantheses
+        """
+        vars = []
+
+        if len(vars_field) == 0:
+            return vars
+
+        list_of_vars = vars_field.split(", ")
+
+        for var in list_of_vars:
+            var_dict = {}
+            sides = var.split(" :: ")
+            var_dict[QM_KEY_TYPE] = sides[1]
+            if " = " in sides[0]:
+                splt = sides[0].split(" = ")
+                var_dict[QM_KEY_NAME] = splt[0]
+                var_dict[QM_KEY_DEFAULT] = splt[1].strip('"')
+            else:
+                var_dict[QM_KEY_NAME] = sides[0]
+
+            vars.append(var_dict)
+
+        return vars
+
+
+class QueryModule:
+    """Class representing a single query module."""
+
+    def __init__(self, module_dict: Dict) -> None:
+        self.name = module_dict[QM_FIELD_NAME]
+        self.is_editable = module_dict[QM_FIELD_IS_EDITABLE]
+        self.is_write = module_dict[QM_FIELD_IS_WRITE]
+        self.path = module_dict[QM_FIELD_PATH]
+        self.signature = module_dict[QM_FIELD_SIGNATURE]
+        self.arguments = module_dict[QM_FIELD_ARGUMENTS]
+        self.returns = module_dict[QM_FIELD_RETURNS]
+
+    def __str__(self) -> str:
+        return self.name
+
+    def set_inputs(self, **kwargs) -> None:
+        """Set values for QueryModule arguments so the module can be called.
+
+        Kwargs:
+            Named arguments in self.arguments.
+
+        Raises:
+            KeyError: Passed an argument not in the self.arguments list.
+        """
+        for kwarg in kwargs.keys():
```

Comment:
I would consider changing variable names, here `kwarg` is not descriptive enough. Maybe `argument_key`.